erasure-readfile.go

```diff
@@ -23,7 +23,7 @@ import (
 )
 
 // ReadFile - decoded erasure coded file.
-func (e erasure) ReadFile(volume, path string, startOffset int64) (io.ReadCloser, error) {
+func (e erasure) ReadFile(volume, path string, startOffset int64, totalSize int64) (io.ReadCloser, error) {
```

Comment:
Make offset as the last argument, with this also change the for loop behavior. So that we don't return back at 'EOF' . So that we can avoid bunch of other unecessary code.